Platon/backend/app/auth_system/helpers.py

```diff
@@ -26,14 +33,13 @@ def auth_check(*args, **kws):
         except:
             return make_response(jsonify({'error' : 'Login Required'}),401)
         try:
-            auth_info = jwt.decode(auth_token,app.config['JWT_SESSION_KEY'],algorithms=[app.config['JWT_ALGORITHM']])
-            expire_time = parser.isoparse(auth_info['expire_time'])
-            if expire_time < datetime.datetime.now():
-                return make_response(jsonify({'error' : 'Expired Token'}),401)
+            id = decode_token(auth_token)
+            if type(id) is not int:
```

Comment:
it confused me, why did you implement type checking? is it because of a bug? 

also, after type checking, only id is returned. shouldn't we also return new token?

Reply:
Because decode_token function retunrns a Respose if token is expired. If it is not expired than it only returns the id of the user. So if the return value is not integer than it means token is expired and we have to return its response instead of continuing to the function.

Reply:
Please examine the login_required more carefully. If you look at it again you will se that this decorator sends new token from the header of the response.